### PR TITLE
Strong name everything

### PR DESCRIPTION
**Pull request by cesarbs**
Signing everything under src. Some tests had to be signed too because they consume internals of some projects in src.

I'm using the key that was already under src/Microsoft.Dnx.Host.Clr. Moved it to tools/Key.snk.

/cc @moozzyk @muratg @davidfowl @pranavkm 

#2615

### CODE REVIEW

File: samples/HelloWorld/project.json [hunk 1]
```diff
@@ -3,7 +3,6 @@
     "exclude": "more/*.cs",
     "dependencies": {
         "HelloShared": { "version": "0.1-beta-*", "type": "build" },
```
Comment (pranavkm) [inline samples/HelloWorld/project.json]: Trailing comma

File: src/Microsoft.Dnx.Compilation.CSharp.Abstractions/project.json [hunk 2]
```diff
@@ -1,5 +1,8 @@
 {
   "version": "1.0.0-*",
+  "compilationOptions": {
+      "keyFile": "../../tools/Key.snk"
```
Comment (pranavkm) [inline src/Microsoft.Dnx.Compilation.CSharp.Abstractions/project.json]: Could you also throw in a `warningAsErrors: true`

File: src/dnx.clr/dnx.clr.vcxproj [hunk 3]
```diff
@@ -16,6 +16,12 @@
...
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
+    <LinkKeyFile>../../tools/Key.snk</LinkKeyFile>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
+    <LinkKeyFile>../../tools/Key.snk</LinkKeyFile>
+  </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
   <Import Project="$(MSBuildThisFileDirectory)..\..\Build\Dnx.Native.Settings" />
```
Comment (pranavkm) [inline src/dnx.clr/dnx.clr.vcxproj]: Could you modify `Dnx.Native.Settings` instead?

Comment (davidfowl) [inline src/dnx.clr/dnx.clr.vcxproj]: Wait what is this for?

Comment (cesarbs) [inline src/dnx.clr/dnx.clr.vcxproj]: I thought that would sign the native dlls.

Comment (moozzyk) [inline src/dnx.clr/dnx.clr.vcxproj]: Nope. Revert any changes to .vcxproj or build/* files.

**Comment by ejsmith** [comment]
I thought the idea was to get rid of the GAC and strong naming with vnext?

**Comment by cesarbs** [comment]
Closing in favor of https://github.com/aspnet/dnx/pull/3061.

**Pull request closed without merging**